webapp/app/[locale]/stake/_components/manageStake/stakeOperation.tsx

```diff
@@ -69,7 +69,12 @@ export const StakeOperation = function ({
   } = useStake(token)
   const t = useTranslations('stake-page.drawer')
   const tCommon = useTranslations('common')
-  const { balance } = useTokenBalance(token)
+  const nativeBalance = useNativeTokenBalance(token.chainId)
+  const tokenBalance = useTokenBalance(token)
+
+  const balance = isNativeToken(token)
+    ? nativeBalance.balance
+    : tokenBalance.balance
```

Comment:
a minor comment, but one way to avoid hitting the complexity is just moving this into a local hook in this file. Like

```tsx
// Create this in this file
const useBalance = function (token) {
  const nativeBalance = useNativeTokenBalance(token.chainId)
  const tokenBalance = useTokenBalance(token)

  const balance = isNativeToken(token)
    ? nativeBalance.balance
    : tokenBalance.balance
}
```

then

```suggestion
  const balance = useBalance(token)
```

Reply:
Done.